### PR TITLE
fix(cli): replace preferred-pm with local implementation

### PR DESCRIPTION
**Pull request by binoy14**
### Description

Replaces the `preferred-pm` npm dependency with a local implementation to fix compatibility with yarn v1 on Node 20. The `preferred-pm@5.0.0` package requires `node >=22.13`, which breaks yarn v1 Classic since it enforces engine checks by default.

See #983 for context on the yarn v1 + Node 20 incompatibility.

Changes:
- Remove `preferred-pm` dependency from `@sanity/cli`
- Add local `preferredPm` module that detects package managers via lock files, workspace configurations, and `node_modules` markers
- Convert from async to sync detection (no behavioral change — the original library only did sync filesystem checks behind an async API)
- Fix workspace detection for paths nested inside a package (e.g. `packages/child/src`)
- Fix `.modules.yaml` parsing to handle both YAML and JSON formats

### What to review

- `packages/@sanity/cli/src/util/packageManager/preferredPm.ts` — the local implementation
- `packages/@sanity/cli/src/util/packageManager/__tests__/preferredPm.test.ts` — unit tests (22 cases)
- `packages/@sanity/cli/src/util/packageManager/packageManagerChoice.ts` — updated import and usage

### Testing

Unit tests cover: lock file detection (all 6 lock file types), parent directory pnpm-lock walk-up, workspace root detection (yarn/npm/pnpm/bun workspaces, subdirectory paths, non-matching paths, `workspaces.packages` format), `node_modules` marker detection (YAML and JSON `.modules.yaml` formats), and empty directory fallback.

```
pnpm test packages/@sanity/cli/src/util/packageManager/__tests__/preferredPm.test.ts
```

---

> [!NOTE]
> **Medium Risk**
> Changes CLI package-manager auto-detection logic and removes a third-party dependency, which can affect dependency installation/upgrade flows across different workspace layouts. Risk is mitigated by extensive new unit tests, but behavior differences in edge cases are still possible.
> 
> **Overview**
> Fixes Yarn v1 on Node 20 by **removing the `preferred-pm` dependency** (and its Node 22+ engine requirement) and switching `getPackageManagerChoice` to use a new in-repo `preferredPm` implementation.
> 
> The new detector infers the preferred package manager from lockfiles, pnpm lock/workspace files in parent directories, workspace root `package.json` globs, and `node_modules` markers (including more tolerant `.modules.yaml` parsing), with a comprehensive new `vitest` suite to cover these cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit a000164b40671ecf76fcb450d583833961c56e5f. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/@sanity/cli/src/util/packageManager/preferredPm.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
+// Based on preferred-pm (MIT) by Zoltan Kochan — https://github.com/zkochan/packages
+// Based on which-pm (MIT) by pnpm — https://github.com/pnpm/which-pm
+
+import fs from 'node:fs'
+import path from 'node:path'
+
+import picomatch from 'picomatch'
+
+import {toForwardSlashes} from '../toForwardSlashes.js'
+import {type PackageManager} from './packageManagerChoice.js'
```
Comment (mttdnt) [inline packages/@sanity/cli/src/util/packageManager/preferredPm.ts]: should this be in a separate types file just to avoid the circular import

File: packages/@sanity/cli/src/util/packageManager/__tests__/preferredPm.test.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+    it('finds pnpm-workspace.yaml in a parent directory', () => {
+      fs.writeFileSync(path.join(tmpDir, 'pnpm-workspace.yaml'), "packages:\n  - 'packages/*'\n")
+      const childDir = path.join(tmpDir, 'packages', 'child')
+      fs.mkdirSync(childDir, {recursive: true})
+      expect(preferredPm(childDir)).toBe('pnpm')
+    })
+  })
+
+  describe('workspace root detection', () => {
```
Comment (mttdnt) [inline packages/@sanity/cli/src/util/packageManager/__tests__/preferredPm.test.ts:74]: In this block maybe add a test for a malformed package.json in detectFromWorkspaceRoot

**Review by mttdnt — DISMISSED** [review]
LGTM, seems close enough to the npm packages, just a couple nits

**Review by mttdnt — APPROVED** [review]
